OPAL/si/src/main/scala/org/opalj/fpcf/scheduling/MaximumPhaseScheduling.scala

```diff
@@ -98,7 +102,60 @@ abstract class MultiplePhaseScheduling extends SchedulingStrategy {
             computePhase(ps, phaseAnalyses, remainingAnalyses)
         }
 
-        schedule
+        if (!config.getBoolean(DisableCleanup)) calculateDeletions(schedule, ps, config) else schedule
+    }
+
+    private def calculateDeletions[A](
+        schedule: List[PhaseConfiguration[A]],
+        ps:       PropertyStore,
+        config:   Config
+    ): List[PhaseConfiguration[A]] = {
+        def keyByName(name: String): SomePropertyKey = {
```

Comment:
We might want to have a more efficient way of turning names into keys implemented in `PropertyKey`, e.g., a map from name to key.